src/weakptr.c

```diff
@@ -163,20 +163,22 @@ Int GrowWPObj (
 ** WP object 
 */
 
-Obj FuncWeakPointerObj( Obj self, Obj list ) { 
+#ifdef BOEHM_GC
+/* We need to make sure that <list> stays live until
+ * after REGISTER_WP(); on architectures that pass
+ * arguments in registers (x86_64, SPARC, etc), the
+ * argument register may be reused. In conjunction with
+ * loop unrolling, the reference to <list> may then be
+ * destroyed before REGISTER_WP() is called.
+ */
+Obj FuncWeakPointerObj( Obj self, volatile Obj list )
```

Comment:
My belief (which I have no evidence for) is adding `(void)list2;` to the old code will force `list2` to be considered "used", which should keep it present, and also stop the warning.